Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs

```diff
@@ -361,6 +366,280 @@ public void OnEndLogsWarningWhenObjectPassedToEndWithoutCorrespondingBeginAsync(
+
+            stopwatch.Stop();
+
+            // wait for OnEnd async completion
+            Thread.Sleep(50);
+
+            Assert.AreSame(returnTaskPassed, objectReturned, "Object returned is not the same as expected return object");
+            Assert.AreEqual(1, this.sendItems.Count, "Only one telemetry item should be sent");
+
+            ValidateTelemetryPacket(
+                this.sendItems[0] as DependencyTelemetry,
+                expectedName: GetResourceNameForStoredProcedure(command),
+                expectedSuccess: false,
+                expectedResultCode: "10",
+                timeBetweenBeginEndInMs: stopwatch.Elapsed.TotalMilliseconds);
+        }
+
+        [TestMethod]
+        public void OnEndAsyncSendsCorrectTelemetry_Exception_2ArgumentsOverride()
+        {
+            var command = GetSqlCommandTestForStoredProc();
+
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            var returnTaskPassed = Task.Factory.StartNew(() => { throw TestUtils.GenerateSqlException(10); });
+            
+            var context = this.sqlCommandProcessingProfiler.OnBeginForTwoParameters(command, null);
+            var objectReturned = this.sqlCommandProcessingProfiler.OnEndAsyncForTwoParameters(context, returnTaskPassed, command, null);
+
+            stopwatch.Stop();
+
+            // wait for OnEnd async completion
+            Thread.Sleep(50);
+
+            Assert.AreSame(returnTaskPassed, objectReturned, "Object returned is not the same as expected return object");
+            Assert.AreEqual(1, this.sendItems.Count, "Only one telemetry item should be sent");
+
+            ValidateTelemetryPacket(
+                this.sendItems[0] as DependencyTelemetry,
+                expectedName: GetResourceNameForStoredProcedure(command),
+                expectedSuccess: false,
+                expectedResultCode: "10",
+                timeBetweenBeginEndInMs: stopwatch.Elapsed.TotalMilliseconds);
+        }
+
+        [TestMethod]
+        public void OnEndAsyncLogsWarningWhenNullTaskPassedToEndMethods()
+        {
+            using (var listener = new TestEventListener())
+            {
+                const long AllKeyword = -1;
+                listener.EnableEvents(DependencyCollectorEventSource.Log, EventLevel.Warning, (EventKeywords)AllKeyword);
+                try
+                {
+                    this.sqlCommandProcessingProfiler.OnEndAsyncForOneParameter(new object(), null, null);
+                }
+                catch (Exception)
+                {
+                    Assert.Fail("Should not be throwing unhandled exceptions.");
+                }
+
+                var message = listener.Messages.First(item => item.EventId == 14);
+                Assert.IsNotNull(message);
+            }
+        }
+
+        [TestMethod]
+        public void OnEndExceptionAsyncNotSendsTelemetryIfSuccess_1ArgumentOverride()
```

Comment:
updated

---
In reply to: [98075481](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/298#discussion_r98075481) [](ancestors = 98075481)